src/app/badges/components/BadgeCard.tsx

```diff
@@ -41,6 +41,10 @@ const BadgeCard: React.FC<BadgeCardProps> = ({
 	medal,
 	amount,
 }) => {
+	const formatAmount = (amount: number | undefined) => {
+		if (amount === undefined) return '';
+		return amount >= 1000000 ? `${amount / 1000000}M` : amount.toString();
```

Comment:
I would say, let's cap the number of decimal places here. Because right now, giving `54_546_784` to the function, it would return `54.546784M` which is not a whole lot more readable.
Something like `${(amount / 1000000).toFixed(2)}M` would be better.